client/asset/eth/eth.go

```diff
@@ -3799,7 +3808,8 @@ func checkTxStatus(receipt *types.Receipt, gasLimit uint64) error {
 // factor of 2. The account should already have a trading balance of at least
```

Comment:
In the `redeem` method just above, I think we need to either use `voidUnusedNonce` if an override is provided, or modify `txOpts()` with an optional nonce override input arg that makes it skip the `nextNonce` call entirely.

Reply:
Moving the check and release up to `*assetWallet.withContractor` methods to cover everything.